crates/runtime/benches/bench_search/datasets.rs

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2024 The Spice.ai OSS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+use app::AppBuilder;
+use hf_hub::{api::tokio::ApiBuilder, Repo, RepoType};
+use spicepod::component::{
+    dataset::{acceleration::Acceleration, Dataset},
+    embeddings::ColumnEmbeddingConfig,
+};
+
+/// The `QuoraRetrieval` MTEB dataset is a benchmark dataset used for evaluating retrieval models.
+/// It consists of 177,163 rows and 1000 test queries.
+/// `https://huggingface.co/datasets/mteb/QuoraRetrieval_test_top_250_only_w_correct-v2/`
+pub(crate) async fn add_mtep_quora_retrieval_dataset(
+    app_builder: AppBuilder,
+    acceleration: Option<Acceleration>,
+) -> Result<AppBuilder, String> {
+    let hf_api = ApiBuilder::new()
```

Comment:
nit: Would be good if we could abstract this as a util function in `crates/llms` so that we don't need to import `hf-hub` into runtime (even if just a dev-dependency).